subprojects/gradle-guides-plugin/src/main/java/org/gradle/docs/samples/internal/SamplesDocumentationPlugin.java

```diff
@@ -415,7 +429,9 @@ private void realizeSamples(SamplesInternal extension, ObjectFactory objects, Ta
                 exemplarBinary.getTestedWorkingDirectory().convention(extension.getTestedInstallRoot().dir(toKebabCase(sample.getName()) + "/" + dsl.getConventionalDirectory())).disallowChanges();
                 exemplarBinary.getTestsContent().from(objects.fileCollection().from((Callable<FileTree>) () -> project.zipTree(binary.getZipFile())).builtBy(binary.getZipFile())); // TODO: zipTree should be lazy
                 exemplarBinary.getTestsContent().from(sample.getSampleDirectory().dir("tests"));
+                exemplarBinary.getDsl().value(dsl);
```

Comment:
WDYT as an alternative:
- Don't add DSL to the install task since we end up using it to filter out content we don't want
- Change the convention to look for conf files in tests/ (not recursively) and conf files in tests/$dsl
- Make sure we flatten these into a single directory when installing to the install directory for test

Reply:
In general, I like a good idea. So, our proposal would end up looking like this:
```
mySample
  |- common
  |- java
  |- kotlin
  |- test
     |- java
     |- kotlin
     |- common.sample.conf
```
How about a flat folder structure? Ultimately a complex sample could look like this:
```
mySample
  |- common
  |- java
  |- kotlin
  |- test-common
  |- test-java
  |- test-kotlin
```

Reply:
> Make sure we flatten these into a single directory when installing to the install directory for test
I don't think the Exemplar tests would otherwise.

Reply:
I've implemented approach.